### PR TITLE
[KStreams FK Joins] Initial import

### PR DESCRIPTION
**Pull request by gunnarmorling**
Hey @rk3rn3r, would you like to review this one? It's a by-product of my Kafka Summit talk, where I'm going to touch on FK joins shortly. So thought I could also provide this as an example here. Thanks!

### CODE REVIEW

**Comment by rk3rn3r** [comment]
Can I do tomorrow? Today was about debugging the Postgres BIT VARYING issue.

**Comment by gunnarmorling** [comment]
Of course, whenever it suits you. Thanks!

>

**Comment by gunnarmorling** [comment]
Hey @rk3rn3r, any thoughts?

File: kstreams-fk-join/aggregator/src/main/docker/Dockerfile.jvm [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# Then, build the image with:
+#
+# docker build -f src/main/docker/Dockerfile.jvm -t quarkus/kafka-streams-quickstart-producer-jvm .
+#
+# Then run the container using:
+#
+# docker run -i --rm -p 8080:8080 quarkus/kafka-streams-quickstart-producer-jvm
+#
+###
+FROM fabric8/java-centos-openjdk8-jdk
```
Comment (rk3rn3r) [inline kstreams-fk-join/aggregator/src/main/docker/Dockerfile.jvm:17]: Maybe using a more recent Java version?

File: kstreams-fk-join/aggregator/src/main/java/io/debezium/examples/kstreams/fkjoin/streams/TopologyProducer.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        
+        KTable<Integer, Customer> customers = builder.table(
+                customersTopic,
+                Consumed.with(customersKeySerde, customersSerde)
+        );
+        
+        KTable<Integer, CustomerWithAddresses> customersWithAddresses = addresses.join(
+                customers,
+                address -> address.customer_id,
+                (address, customer) -> new AddressAndCustomer(address, customer),
```
Comment (rk3rn3r) [inline kstreams-fk-join/aggregator/src/main/java/io/debezium/examples/kstreams/fkjoin/streams/TopologyProducer.java:69]: can be replaced with lambda  `AddressAndCustomer::new`

File: kstreams-fk-join/aggregator/src/main/docker/Dockerfile.jvm (hunk 1 above)
Comment (gunnarmorling) [reply kstreams-fk-join/aggregator/src/main/docker/Dockerfile.jvm:17]: Shouldn't matter too much for these purposes here.

File: kstreams-fk-join/aggregator/src/main/java/io/debezium/examples/kstreams/fkjoin/streams/TopologyProducer.java (hunk 2 above)
Comment (gunnarmorling) [reply kstreams-fk-join/aggregator/src/main/java/io/debezium/examples/kstreams/fkjoin/streams/TopologyProducer.java:69]: I had tried that, but it doesn't compile. Not quite sure why tbh., I was expecting it to work, too. In any case not worth spending much effort.

**Comment by gunnarmorling** [comment]
@rk3rn3r, unless you have other remarks, could you rebase and merge it? Nice excercise to see whether your permissions are right? Thanks!

**Comment by rk3rn3r** [comment]
Worked. @gunnarmorling

**Pull request merged**